crates/core/src/globals.rs

```diff
@@ -255,6 +272,60 @@ fn build_config_object(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
     Ok(config_obj)
 }
 
+fn build_memory(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
+    let memory_from_buffer = context.wrap_callback(
+        |_ctx: &JSContextRef, _this: JSValueRef, args: &[JSValueRef]| {
+            let data = args.get(0).ok_or(anyhow!("Expected data argument"))?;
+            if !data.is_array_buffer() {
+                bail!("Expected data to be an array buffer");
+            }
+            let data = data.as_bytes()?;
+            let m = extism_pdk::Memory::from_bytes(data)?;
+            let mut mem = HashMap::new();
+            let offset = JSValue::Int(m.offset() as i32);
+            let len = JSValue::Int(m.len() as i32);
+            mem.insert("offset".to_string(), offset);
+            mem.insert("len".to_string(), len);
+            Ok(JSValue::Object(mem))
+        },
+    )?;
+    let memory_find = context.wrap_callback(
+        |_ctx: &JSContextRef, _this: JSValueRef, args: &[JSValueRef]| {
+            let ptr = args.get(0).ok_or(anyhow!("Expected ptr argument"))?;
+            if !ptr.is_number() {
+                bail!("Expected a pointer");
+            }
+            let ptr = ptr.as_i32_unchecked();
+            let m = extism_pdk::Memory::find(ptr as u64).unwrap();
+            let mut mem = HashMap::new();
+            let offset = JSValue::Int(m.offset() as i32);
+            let len = JSValue::Int(m.len() as i32);
+            mem.insert("offset".to_string(), offset);
+            mem.insert("len".to_string(), len);
+            Ok(JSValue::Object(mem))
+        },
+    )?;
+    let read_bytes = context.wrap_callback(
+        |_ctx: &JSContextRef, _this: JSValueRef, args: &[JSValueRef]| {
+            let ptr = args.get(0).ok_or(anyhow!("Expected ptr argument"))?;
+            if !ptr.is_number() {
+                bail!("Expected a pointer");
+            }
+            let ptr = ptr.as_i32_unchecked();
+            let m = extism_pdk::Memory::find(ptr as u64).unwrap();
+            let bytes = m.to_vec();
+            Ok(JSValue::ArrayBuffer(bytes))
+        },
+    )?;
+
+    let mem_obj = context.object_value()?;
+    mem_obj.set_property("fromBuffer", memory_from_buffer)?;
+    mem_obj.set_property("find", memory_find)?;
+    mem_obj.set_property("readBytes", read_bytes)?;
```

Comment:
Going tot think about this interface a little bit more

Reply:
Note: I think we should add a free as well. Need to reference the other PDK APIs and try to match it.